### PR TITLE
Add `ruff` to `pre-commit` hooks

### PR DESCRIPTION
**Pull request by clot27**
Added `ruff` to `pre-commit` hooks, changed some files because ruff was complaining about them. Not sure what to add in its args..

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -79,3 +79,9 @@ repos:
         args:
         - --diff
         - --check
+-   repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 'v0.0.212'
+    hooks:
+    -   id: ruff
+        name: ruff
+        files: ^(telegram|examples|tests)/.*\.py$
```
Comment (Bibo-Joshi) [inline .pre-commit-config.yaml:87]: new line is missing :)

File: pyproject.toml [hunk 2]
```diff
@@ -5,3 +5,6 @@ target-version = ['py37', 'py38', 'py39', 'py310', 'py311']
 [tool.isort]  # black config
 profile = "black"
 line_length = 99
+
+[tool.ruff]
+line-length = 99
```
Comment (Bibo-Joshi) [inline pyproject.toml:10]: new line at the end of file is missing

File: tests/test_updater.py [hunk 3]
```diff
@@ -532,7 +532,7 @@ async def test_webhook_basic(
         # that depends on this distinction works
         if ext_bot and not isinstance(updater.bot, ExtBot):
             updater.bot = ExtBot(updater.bot.token)
-        if not ext_bot and not type(updater.bot) is Bot:
+        if not ext_bot and not isinstance(updater.bot, Bot):
```
Comment (Bibo-Joshi) [inline tests/test_updater.py:535]: mh, here want to ensure that we don't have an `ExtBot` instance though … So either we tell ruff to ignore that or we use `and not isinstance(updater.bot, ExtBot)`

or we just don't run ruff on the test files. IISC we also don't run `mypy` and `pylint` on those …

**Review by Bibo-Joshi — COMMENTED** [review]
thanks for the PR :)

File: tests/test_updater.py (hunk 3 above)
Comment (clot27) [reply tests/test_updater.py:535]: Oke, I will remove ruff to check tests

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -79,3 +79,15 @@ repos:
...
+    hooks:
+    -   id: ruff
+        name: ruff
+        files: ^(telegram|examples)/.*\.py$
+        additional_dependencies:
+          - httpx~=0.23.0
+          - tornado~=6.2
+          - APScheduler~=3.9.1
+          - cachetools~=5.2.0
+          - aiolimiter~=1.0.0
```
Comment (Bibo-Joshi) [inline .pre-commit-config.yaml:93]: new line missing 🙃

Comment (clot27) [reply .pre-commit-config.yaml:93]: Uh, ok

**Pull request merged**